test/functional/session_controller_test.rb

```diff
@@ -0,0 +1,118 @@
+require "test_helper"
+require "gds_api/test_helpers/account_api"
+
+class SessionsControllerTest < ActionController::TestCase
+  include GdsApi::TestHelpers::AccountApi
+  include SessionHelpers
+
+  context "GET sign-in" do
+    setup do
+      stub_account_api_get_sign_in_url(redirect_path: "/bank-holiday")
+    end
+
+    should "redirect the user to the GOV.UK Account service domain" do
+      get :create, params: { redirect_path: "/bank-holiday", state_id: "abc123" }
+      assert_response :redirect
+      assert_equal @response.headers["Location"], "http://auth/provider"
+    end
+
+    should "preserve the _ga tracking parameter if provided" do
+      get :create, params: { redirect_path: "/bank-holiday", state_id: "abc123", _ga: "abc123" }
+      assert_equal @response.headers["Location"], "http://auth/provider?_ga=abc123"
+    end
+
+    context "when already logged in" do
+      setup { mock_logged_in_session }
+
+      should "redirect the user to the account manager if the user is already logged in" do
+        get :create, params: { redirect_path: "/bank-holiday", state_id: "abc123" }
+        assert_redirected_to Plek.find("account-manager")
+      end
+    end
+  end
+
+  context "GET sign-in/callback" do
+    setup do
+      @govuk_account_session = "govuk_account_session"
```

Comment:
This instance variable is only used in a single test, how about inlining the variable there?